### DISCUSSION TITLE
Adaptation of the authentication flow

### DISCUSSION THREAD

**Discussion by Konst80** [serverpod/serverpod#1345] [General]
Hi everyone,

I'm looking to adapt the authentication flow to meet my requirements. However, I haven't found documentation on how to achieve the following:

- Registration without a confirmation code
- Transmitting additional parameters from the client to the server during registration
- Executing custom code on the server during registration
- Implementing additional functions, such as .deleteAccount()

Does the serverpod_auth_server / _client plugin support these adaptations at a conceptual level? Or is it the case that users need to create their own authentication module?

**Comment by Frank1016** [comment]
As far as I know, the current auth module from Serverpod won't meet the requirements listed above. Probably, you need to create your own auth flow for more customization.

**Reply by Konst80** [reply]
Thank you Frank, 

I have already suspected the same thing. 

After I have looked at the code of the plugin, I am now also sure that these requirements can not be covered by the plugin itself. But it can be modified accordingly...I'll try ;-)
